universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramsController.java

```diff
@@ -0,0 +1,38 @@
+package controllers.applicant;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import play.mvc.Controller;
+import play.mvc.Result;
+import services.program.ProgramService;
+import views.applicant.ProgramIndexView;
+
+/** Controller for handling methods for an applicant applying to programs. */
+public class ApplicantProgramsController extends Controller {
+
+  private final ProgramService programService;
+  private final ProgramIndexView programIndexView;
+
+  @Inject
+  public ApplicantProgramsController(
+      ProgramService programService, ProgramIndexView programIndexView) {
+    this.programService = checkNotNull(programService);
+    this.programIndexView = checkNotNull(programIndexView);
+  }
+
+  public CompletionStage<Result> index(long applicantId) {
+    return programService
+        .listProgramDefinitionsAsync()
+        .thenApply(programs -> ok(programIndexView.render(applicantId, programs)));
```

Comment:
Suggestion (blocking): when continuing from async calls like this the play docs recommend explicitly specifying the `HttpExecutionContext` to ensure that the class loader is available in its correct state.

```java
// inject it in the c'tor
@Inject
ApplicantProgramsController(HttpExecutionContext httpExecutionContext)
```
then
```java
         .thenApply(programs -> ok(programIndexView.render(applicantId, programs)),
           httpExecutionContext.current()); 
```

Reply:
Thanks for linking the docs, I completely missed that section